spring-integration-core/src/main/java/org/springframework/integration/endpoint/PollingConsumer.java

```diff
@@ -68,18 +87,82 @@ protected void doStop() {
 		super.doStop();
 	}
 
-
 	@Override
 	protected void handleMessage(Message<?> message) {
-		this.handler.handleMessage(message);
+		Deque<ExecutorChannelInterceptor> interceptorStack = null;
+		try {
+			if (this.channelInterceptors != null
+					&& ((ExecutorChannelInterceptorAware) this.inputChannel).hasExecutorInterceptors()) {
+				interceptorStack = new ArrayDeque<ExecutorChannelInterceptor>();
+				message = applyBeforeHandle(message, interceptorStack);
+				if (message == null) {
+					return;
+				}
+			}
+			this.handler.handleMessage(message);
+			if (!CollectionUtils.isEmpty(interceptorStack)) {
+				triggerAfterMessageHandled(message, null, interceptorStack);
+			}
+		}
+		catch (Exception ex) {
+			if (!CollectionUtils.isEmpty(interceptorStack)) {
+				triggerAfterMessageHandled(message, ex, interceptorStack);
+			}
+			if (ex instanceof MessagingException) {
+				throw (MessagingException) ex;
+			}
+			String description = "Failed to handle " + message + " to " + this + " in " + this.handler;
+			throw new MessageDeliveryException(message, description, ex);
+		}
+		catch (Error ex) {
```

Comment:
Add `//NOSONAR - ok because we re-throw it below`

Otherwise Sonar will bark at us.